### PR TITLE
Transfer in tips

### PR DESCRIPTION
**Pull request by ElizabethSamuel-MSFT**
Related to internal task 4989446.

### CODE REVIEW

File: docs/testing/tips.md [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+* `range.setValues([['Hello', 'World!]])` --> this will set two adjacent cells of the range. If you are using a plural version of the API, the dimension of the range should match the dimension of the argument.
+
+## Other miscellaneous tips
+
+## Full screen editing
+
+A full-screen detachable editor is not yet offered in Office Scripts. In the meantime, you can use the following steps to get code completion in Visual Studio Code. It's only for editing; you can't execute a script using this. To run the script, you still have to copy/paste code into the Office Scripts in-built code editor.
+
+1. In case you already don't have it, download and install the [Visual Studio Code editor](https://code.visualstudio.com/).
+1. Download the intelliSense (code completion assistance) [file](index.d.ts) to your local machine into a base folder from where you'll edit your script files.
```
Comment (ElizabethSamuel-MSFT) [inline docs/testing/tips.md:242]: Where should we grab index.d.ts from? Or should we have user just point directly to DT or office-scripts copy?

Comment (ElizabethSamuel-MSFT) [inline docs/testing/tips.md:242]:
```suggestion
1. Download the IntelliSense (code completion assistance) [file](index.d.ts) to your local machine into a base folder from where you'll edit your script files.
```

File: docs/testing/tips.md [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+* `range.setValues([['Hello', 'World!]])` --> this will set two adjacent cells of the range. If you are using a plural version of the API, the dimension of the range should match the dimension of the argument.
+
+## Other miscellaneous tips
+
+## Full screen editing
+
+A full-screen detachable editor is not yet offered in Office Scripts. In the meantime, you can use the following steps to get code completion in Visual Studio Code. It's only for editing; you can't execute a script using this. To run the script, you still have to copy/paste code into the Office Scripts in-built code editor.
+
+1. In case you already don't have it, download and install the [Visual Studio Code editor](https://code.visualstudio.com/).
+1. Download the IntelliSense (code completion assistance) [file](index.d.ts) to your local machine into a base folder from where you'll edit your script files.
```
Comment (ElizabethSamuel-MSFT) [inline docs/testing/tips.md:242]: Where should we grab index.d.ts from? Or should we have user just point directly to DT or office-scripts copy?

Comment (AlexJerabek) [reply docs/testing/tips.md:242]: The type definitions aren't in DefinitelyTyped. I think the only public d.ts file for Scripts is in our ref docs. Tagging @sumurthy to advise.

**Comment by AlexJerabek** [comment]
@ElizabethSamuel-MSFT and @alison-mk, I'd like your thoughts on how to progress with the information in this PR. A lot of the content has been merged with the docs across other, smaller PRs and I think we're done, but I want to check with you. Here's the status of each section:

1. (IntelliSense tips) - In PR #343.
2. (type assertion with Ranges) - In PR #334.
3. (method parameter types) - This feels like TypeScript 101. I added a [section about TypeScript](https://docs.microsoft.com/en-us/office/dev/scripts/develop/scripting-fundamentals#typescript-the-language-of-office-scripts) to the fundamentals page, but I don't think specifics about types belongs here.
4. (null or undefined) - Added a [section about that](https://docs.microsoft.com/en-us/office/dev/scripts/develop/scripting-fundamentals#verify-an-object-exists-in-the-collection) to the fundamentals page. I omitted the `?` because it felt both advanced and not Scripts-specific.
5. (using the map class) - While this is a cool tip, I'm not sure I see the real-world value. There are other ways to handle 2D arrays and we show them in numerous samples. The map function is slick, but feels a little too advanced for the audience.
6. (using VS Code) - This doesn't feel like something we'd recommend to our audience. It's a pro-dev thing that could make an interesting blog post, but I think we want makers staying in Excel.

**Comment by ElizabethSamuel-MSFT** [comment]
> @ElizabethSamuel-MSFT and @alison-mk, I'd like your thoughts on how to progress with the information in this PR. A lot of the content has been merged with the docs across other, smaller PRs and I think we're done, but I want to check with you. Here's the status of each section:
> 
> 1. (IntelliSense tips) - In PR #343.
> 2. (type assertion with Ranges) - In PR #334.
> 3. (method parameter types) - This feels like TypeScript 101. I added a [section about TypeScript](https://docs.microsoft.com/en-us/office/dev/scripts/develop/scripting-fundamentals#typescript-the-language-of-office-scripts) to the fundamentals page, but I don't think specifics about types belongs here.
> 4. (null or undefined) - Added a [section about that](https://docs.microsoft.com/en-us/office/dev/scripts/develop/scripting-fundamentals#verify-an-object-exists-in-the-collection) to the fundamentals page. I omitted the `?` because it felt both advanced and not Scripts-specific.
> 5. (using the map class) - While this is a cool tip, I'm not sure I see the real-world value. There are other ways to handle 2D arrays and we show them in numerous samples. The map function is slick, but feels a little too advanced for the audience.
> 6. (using VS Code) - This doesn't feel like something we'd recommend to our audience. It's a pro-dev thing that could make an interesting blog post, but I think we want makers staying in Excel.

Skimmed through the points and referenced PRs/pages, and this seems reasonable to me.

**Comment by AlexJerabek** [comment]
Closing the PR based on the above discussion.

**Comment by alison-mk** [comment]
> @ElizabethSamuel-MSFT and @alison-mk, I'd like your thoughts on how to progress with the information in this PR. A lot of the content has been merged with the docs across other, smaller PRs and I think we're done, but I want to check with you. Here's the status of each section:
> 
> 1. (IntelliSense tips) - In PR #343.
> 2. (type assertion with Ranges) - In PR #334.
> 3. (method parameter types) - This feels like TypeScript 101. I added a [section about TypeScript](https://docs.microsoft.com/en-us/office/dev/scripts/develop/scripting-fundamentals#typescript-the-language-of-office-scripts) to the fundamentals page, but I don't think specifics about types belongs here.
> 4. (null or undefined) - Added a [section about that](https://docs.microsoft.com/en-us/office/dev/scripts/develop/scripting-fundamentals#verify-an-object-exists-in-the-collection) to the fundamentals page. I omitted the `?` because it felt both advanced and not Scripts-specific.
> 5. (using the map class) - While this is a cool tip, I'm not sure I see the real-world value. There are other ways to handle 2D arrays and we show them in numerous samples. The map function is slick, but feels a little too advanced for the audience.
> 6. (using VS Code) - This doesn't feel like something we'd recommend to our audience. It's a pro-dev thing that could make an interesting blog post, but I think we want makers staying in Excel.

Sounds good to me, too.

**Pull request closed without merging**